src/Testcontainers/Builders/DockerRegistryAuthenticationProvider.cs

```diff
@@ -84,7 +84,11 @@ private IDockerRegistryAuthenticationConfiguration GetUncachedAuthConfig(string
           {
             authConfig = new IDockerRegistryAuthenticationProvider[]
               {
-                new CredsHelperProvider(dockerConfigDocument, this.logger), new CredsStoreProvider(dockerConfigDocument, this.logger), new Base64Provider(dockerConfigDocument, this.logger),
+                // environment variable provider goes first as it should overwrite the Docker configuration file when set
+                new EnvironmentVariableBase64Provider(this.logger),
+                new CredsHelperProvider(dockerConfigDocument, this.logger),
+                new CredsStoreProvider(dockerConfigDocument, this.logger),
+                new Base64Provider(dockerConfigDocument, this.logger),
```

Comment:
If you extend `CustomConfiguration` as suggested below, you can pass the `GetDockerAuthConfig` result to `Base64Provider`. I think you don't need `EnvironmentVariableBase64Provider` at all.